### ISSUE TITLE
Setting search no longer holding user back from typing non-existing setting

### ISSUE THREAD

**Issue by Kuina-chan** [ppy/osu#24573]
### Type

Game behaviour

### Bug description

I'm testing for a ruleset at the 1920x1057 resolution, after I finish I want to go back to windowed, 720 on the height but it do 768 instead. I went to find a setting for resolution in the setting menu but found no. Suspicious of what went wrong, I type randomly and it just letting me type without holding me back with a a little warning type.

### Screenshots or videos

https://github.com/ppy/osu/assets/75318520/140a7b30-e8f6-4838-af85-1a7fbb74336d

### Version

2023.815.0

### Logs

[database.log](https://github.com/ppy/osu/files/12362945/database.log)
[input.log](https://github.com/ppy/osu/files/12362946/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/12362947/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/12362948/network.log)
[performance.log](https://github.com/ppy/osu/files/12362949/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12362950/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12362951/updater.log)

**Comment by peppy** [comment]
You're comparing to stable behaviour, correct?

**Comment by Kuina-chan** [comment]
No, this is a behaviour I also see in the previous updates, but now it just gone.

**Comment by peppy** [comment]
That's weird, because as far as I can tell it was never implemented.

**Comment by yesseruser** [comment]
I'd like to take a look. Please don't assign me before I confirm I'd like to work on it.

**Comment by Gooeys1** [comment]
Probably duplicate of #4252

**Comment by Joehuu** [comment]
Seems like it.
